### DISCUSSION TITLE
Frigate+ questions

### DISCUSSION THREAD

**Discussion by stalakerob** [blakeblackshear/frigate#11988] [Ask A Question] [answered]
### What is your question:

I was reading the Frigate+ docs and searched the forum here but some things are not perfectly clear to me. I'm trying to understand:

- Can I create my own, totally new objects or does Frigate+ "only" improve the existing objects in my environment?
- Why do I need to upload 10 labelled pictures first before I can use the other features?
- Why can I only label those 10 pictures with a set of predefined labels and not any existing label I'm interested in? For example I cannot label a bird object.
- Does Frigate+ provide access to any new object models from other users that are not included in the set of default objects?

Thanks in advance.
Regards Robert

**Comment by blakeblackshear** [comment, marked as answer]
> Can I create my own, totally new objects or does Frigate+ "only" improve the existing objects in my environment?

The models only support a specific defined set of objects. You can't add custom objects nor does it support all of the labels available in the default model. Here is a quote from the docs:

"Frigate+ models support a more relevant set of objects for security cameras. Currently, only the following objects are supported: person, face, car, license_plate, amazon, ups, fedex, package, dog, cat, deer. Other object types available in the default Frigate model are not available. Additional object types will be added in future releases."

> Why do I need to upload 10 labelled pictures first before I can use the other features?

This is because many users have cameras that look different enough that the base model won't perform well without some tuning. I don't want people to think they can just use the base model and get good results. Another quote from the docs:

"The baseline model isn't directly available after subscribing. This may change in the future, but for now you will need to submit a model request with the minimum number of images."

> Why can I only label those 10 pictures with a set of predefined labels and not any existing label I'm interested in? For example I cannot label a bird object.

I picked a limited set of objects to start with and will be adding more in the future. I didn't want to start out by having users label objects that wouldn't be supported in the models they requested.

> Does Frigate+ provide access to any new object models from other users that are not included in the set of default objects?

I don't understand the question. Every model you request in Frigate+ will support all of the labels listed above. From the docs:

"The images you upload are used to fine tune a baseline model trained from images uploaded by all Frigate+ users. This fine tuning process results in a model that is optimized for accuracy in your specific conditions."

and

"This model will support all [label types available](https://docs.frigate.video/plus/#available-label-types) even if you do not submit any examples for those labels."

**Reply by stalakerob** [reply]
Thanks a lot for your clarifications. Do you plan to allow users to add custom objects anytime in the future?

**Reply by blakeblackshear** [reply]
I would like to. It will require a lot of changes, so for now I am focused on making it easier to expand a common set of labels. Custom objects will also require a lot more examples. You might need to provide hundreds or thousands of examples to get good results for something since you don't the collective benefit of pooling examples across users.

**Comment by heisenberg2980** [comment]
I have another question about Frigate+ (please let me know if I should open a new discussion for this, or if it has already been answered): is there a limit on the number of cameras, or the images I upload and label per camera, when creating a model?

**Reply by NickM-27** [reply]
No, there are no camera or image limits (as a max). The only limits are that you must upload _at least_ 10 images between each model
